### PR TITLE
Use setup-emsdk instead of emscripten dockerfile in samples.yml.

### PR DESCRIPTION
**Pull request by ScottTodd**
Also delete the now unused `emscripten.Dockerfile` (it's technically still referenced in a commented out `cross_compile_and_test` build, but that can be added back as needed using this same technique).

Progress on https://github.com/iree-org/iree/issues/15332 - one less Dockerfile to maintain.

Tested here: https://github.com/ScottTodd/iree/actions/runs/10691087132/job/29636852886 (this samples.yml workflow runs on a nightly schedule).

skip-ci: no impact on other workflows

### CODE REVIEW

**Review by saienduri — APPROVED** [review]
lgtm

File: .github/workflows/samples.yml [hunk 1]
```diff
@@ -61,7 +61,6 @@ jobs:
 
   web:
     runs-on: ubuntu-20.04
-    container: gcr.io/iree-oss/emscripten@sha256:2dd4c52f1bb499ab365aad0111fe5538b685d88af38636b409b0cf6a576ab214
     env:
       VENV_DIR: ${{ github.workspace }}/.venv
```
Comment (saienduri) [inline .github/workflows/samples.yml:65]: might be cleaner to add env variables here in future btw

Comment (ScottTodd) [reply .github/workflows/samples.yml:65]: Trueee... I could simplify the various places this is done, like https://github.com/iree-org/iree/blob/3bdd90912ddd5783dd4346dfec372f0d3107f7bd/.github/workflows/pkgci_test_android.yml#L64-L69

File: build_tools/docker/dockerfiles/emscripten.Dockerfile [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-#   (B) Mount a user home directory when using the image.
-# Since (A) requires less configuration, we'll do that. If multiple tools would
-# want a user directory (like Bazel), we should switch to (B).
-# See https://github.com/emscripten-core/emsdk/issues/535
-RUN mkdir -p "${EM_CACHE}" && chmod -R 777 "${EM_CACHE}"
-
-# Normally we'd run `source emsdk_env.sh`, but that doesn't integrate with
-# Docker's environment properties model. Instead, we directly extend the path
-# to include the directories suggested by `emsdk activate`.
-ENV PATH="${EMSDK}:${EMSDK}/upstream/emscripten:$PATH"
```
Comment (ScottTodd) [inline build_tools/docker/dockerfiles/emscripten.Dockerfile:46]: Good riddance to these hacks 🥳

File: .github/workflows/samples.yml (hunk 1 above)
Comment (marbre) [reply .github/workflows/samples.yml:65]: Do we prefer one over the other? Most workflows I've looked in seem to follow https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#defining-environment-variables-for-a-single-workflow and not https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable but I don't know what we should prefer.

**Pull request merged**